### PR TITLE
Some more improvements to Spark test and development setup

### PR DESCRIPTION
**Pull request by amCap1712**
Currently, the hadoop image used during test setup is based on https://github.com/metabrainz/hadoop-cluster-docker. That image is based on older versions of Java and Hadoop than we use in production. Also, removed a couple of files we no longer need. Now that we do not need the old images hadoop images, we can archive the repository at https://github.com/metabrainz/hadoop-cluster-docker.

The local development setup is also modified to use override the test setup docker compose file.

### CODE REVIEW

**Review by alastair — COMMENTED** [review]
I ran the instructions at https://listenbrainz.readthedocs.io/en/production/dev/spark-devel-env/ and everything worked, so looks good to me.

One thing I noticed, but slightly unrelated to this PR, at 
https://listenbrainz.readthedocs.io/en/production/dev/spark-architecture/#start-the-spark-reader the command `./develop.sh manage spark request_user_stats` is incorrect, it needs some arguments. It would be good if we can make sure that all commands given here work with just copy/paste

File: docker/docker-compose.spark.yml [hunk 1]
```diff
@@ -4,35 +4,31 @@ version: "3.4"
 # directory that this file is in (`docker/`) and so probably need to start with
 # `../` to refer to a directory in the main code checkout
```
[the remark this answers is not in the document]
Comment (amCap1712) [reply docker/docker-compose.spark.yml:5]: I have mentioned in the override file.

**Review by alastair — APPROVED** [review]
Thanks!

**Comment by alastair** [comment]
Does this fix LB-873?

**Comment by amCap1712** [comment]
Yes, it does.

**Pull request merged**